controllers/prep_handlers.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	ranv1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func (r *ImageBasedUpgradeReconciler) handlePrep(ctx context.Context, ibu *ranv1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+
+	// TODO actual steps
+	// If completed, update conditions and return doNotRequeue
+	utils.SetStatusCondition(&ibu.Status.Conditions,
+		utils.GetCompletedConditionType(ranv1alpha1.Stages.Prep),
+		utils.ConditionReasons.Completed,
+		metav1.ConditionTrue,
+		"Prep completed",
+		ibu.Generation)
+	utils.SetStatusCondition(&ibu.Status.Conditions,
+		utils.GetInProgressConditionType(ranv1alpha1.Stages.Prep),
+		utils.ConditionReasons.Completed,
+		metav1.ConditionFalse,
+		"Prep completed",
+		ibu.Generation)
```

Comment:
The Idle condition is updated with reason "InProgress" prior to processing each stage. Should it be updated to "Completed" in all completed cases?

Reply:
The idle condition is for the whole IBU. It doesn't go back to true until upgrade/rollback/abort is done.

Reply:
Right, the idle status should still stay in false, but what about the condition reason?  Previously, it was set to Inprogress with desired stage, should it be updated to Completed after the desired stage is completed?

				utils.SetStatusCondition(&ibu.Status.Conditions,
					utils.ConditionTypes.Idle,
					utils.ConditionReasons.InProgress,
					metav1.ConditionFalse,
					string(utils.GetInProgressConditionType(desiredStage)),
					ibu.Generation)